### PR TITLE
Fix musl+static switches for Alpine

### PR DESCRIPTION
**Pull request by XVilka**
Trying to fix https://github.com/ocaml/ocaml/issues/9390 in a "right way".
For now limited with only one compiler version, if everything is fine - I will add the rest.

### CODE REVIEW

File: packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam [hunk 1]
```diff
@@ -32,7 +33,14 @@ build: [
...
     "LIBS=-static"
-  ] {os = "openbsd" | os = "freebsd" | os = "macos" | os = "linux"}
+  ] {os = "openbsd" | os = "freebsd" | os = "macos" | os = "linux" & os-distribution != "alpine" }
+  [
+    "./configure"
+    "--prefix=%{prefix}%"
+    "CFLAGS=-Os"
+    "--enable-flambda"
+    "LIBS=-static"
+  ] {os = "linux" & os-distribution != "alpine" }
```
Comment (gasche) [inline packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam]: You are checking `!= alpine` in both cases, so which configure is supposed to run for alpine then?

Comment (XVilka) [reply packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam]: Thanks, was obviously a typo, fixed.

**Comment by mseri** [comment]
Thanks, Cc also @avsm

**Comment by avsm** [comment]
This needs a manual test since CI won't run it (and the typo in the original PR indicates a lack of previous testing).

File: packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam [hunk 2]
```diff
@@ -17,6 +17,7 @@ depends: [
 depexts: [
   ["musl-tools"] {os-family = "debian"}
   ["musl"] {os-distribution = "arch"}
+  ["musl-dev"] {os-distribution = "alpine"}
```
Comment (avsm) [inline packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam]: Is this really needed on Alpine? musl is the default libc on Alpine, and it compiled before without this depext.

Comment (XVilka) [reply packages/ocaml-variants/ocaml-variants.4.10.0+musl+static+flambda/opam]: musl-dev contains the *.a libc required for the static linking: https://pkgs.alpinelinux.org/contents?branch=edge&name=musl-dev&arch=x86_64&repo=main

**Comment by mseri** [comment]
Any update on this?

**Comment by XVilka** [comment]
I can proceed if this approach is good

**Comment by avsm** [comment]
This broadly looks fine to me, but I'm a bit wary of adding `-Os` to the CFLAGS. It's yet another deviation that may cause unexpected behaviour. Is it necessary?

**Comment by XVilka** [comment]
@avsm it was here before. It's not necessary, but for some reason it was here already.

**Comment by avsm** [comment]
Indeed, we can fix that in a future PR.  This can be merged, or you can do one more simplification -- you don't need to duplicate the build stanzas, as you can put the constraint inline to just the portion of the configure script that is alpine-specific (see the build stanza just below it, where the -j flag is dropped on Cygwin alone.  That'll prevent any future cut-and-paste deviation from Alpine and the other distributions.

**Comment by mseri** [comment]
@XVilka I have simplified the build command. Can you test this one? If it is ok I'll be happy to merge this and the subsequent PR for the musl+static variants.

**Comment by mseri** [comment]
I made a couple of local tests that seem to work fine. I'll merge this one. @XVilka once you have time it will be good to also have the other ones. Thanks again all for the help

**Comment by XVilka** [comment]
I am sorry, was (and still is) very busy with other things. Will send the rest ASAP once back to this.

**Comment by mseri** [comment]
No hurry! Thanks for the help

**Pull request merged**